### PR TITLE
[Concept Entry] Sklearn: Stochastic Gradient Descent

### LINKED ISSUE #4861: [Concept Entry] Sklearn stochastic-gradient-descent
### Reason/inspiration (optional)

We would like a new entry on the `Stochastic Gradient Descent` concept under Sklearn. The entry should go in a new file under `docs/content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept (if applicable)
- An `Example` section that provides an example demonstrating the concept in use
- A `Codebyte Example` section that provides a codebyte example demonstrating the concept in use

Please refer to the [concept entry template](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Concept Entrty for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Sriparno08**
### Description

Added a new concept entry on `Stochastic Gradient Descent` in Sklearn.

### Issue Solved

Closes #4861

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+  - 'Machine Learning'
+  - 'Scikit-learn'
+  - 'Supervised Learning'
+  - 'Unsupervised Learning'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+In Sklearn, **Stochastic Gradient Descent (SGD)** is a popular optimization algorithm that focuses on finding the best set of parameters for a model that minimizes a given loss function. Unlike traditional gradient descent, which calculates the gradient using the entire dataset, SGD computes the gradient using a single training example at a time. This makes it computationally efficient for large datasets.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md:17]: We can add relevant links here and also break it into two parts for better readability - 

```suggestion
In [Sklearn](https://www.codecademy.com/resources/docs/sklearn), **Stochastic Gradient Descent (SGD)** is a popular optimization algorithm that focuses on finding the best set of parameters for a model that minimizes a given loss function. 

Unlike traditional [gradient descent](https://www.codecademy.com/resources/docs/ai/search-algorithms/gradient-descent), which calculates the gradient using the entire dataset, SGD computes the gradient using a single training example at a time. This makes it computationally efficient for large datasets.
```

File: content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+Sklearn provides two primary classes for implementing SGD:
+
+- `SGDClassifier`: Well-suited for classification tasks. Supports various loss functions and penalties for fitting linear classification models.
+- `SGDRegressor`: Well-suited for regression tasks. Supports various loss functions and penalties for fitting linear regression models.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md:22]: Add relevant links wherever possible -

```suggestion
- `SGDRegressor`: Well-suited for regression tasks. Supports various loss functions and penalties for fitting [linear regression models](https://www.codecademy.com/resources/docs/sklearn/linear-regression-analysis).
```

File: content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+# Fit the regressor to the training data
+model.fit(X_train, y_train)
+
+# Make predictions on the new data
+y_pred = model.predict(X_test)
+```
+
+- `loss`: Specifies the loss function.
+  - For `SGDClassifier`, the options include `"hinge"` (default), `"log"`, and `"modified_huber"`.
+  - For `SGDRegressor`, the options include `"squared_loss"` (default), `"huber"`, and `"epsilon_insensitive"`.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md:58]: We should only keep the exact coding term or keyword in backticks -

```suggestion
  - For `SGDClassifier`, the options include `hinge` (default), `log`, and `modified_huber`.
  - For `SGDRegressor`, the options include `squared_loss` (default), `huber`, and `epsilon_insensitive`.
```

File: content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+- `penalty`: Specifies the regularization penalty. Common options include `"l2"` (L2 regularization, default), `"l1"` (L1 regularization), and `"elasticnet"` (a combination of L1 and L2 regularization).
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md:59]: We should only keep the exact coding term or keyword in backticks -

```suggestion
- `penalty`: Specifies the regularization penalty. Common options include `l2` (L2 regularization, default), `l1` (L1 regularization), and `elasticnet` (a combination of L1 and L2 regularization).
```

File: content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+- `max_iter`: Specifies the maximum number of iterations for the optimization algorithm. The default value is `1000`.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md:60]: The description of `max_iter` does not clarify its interaction with convergence. It might imply that increasing it always improves performance, which is not necessarily true.


```suggestion
- `max_iter`: Specifies the maximum number of iterations for the optimization algorithm. The default value is `1000`. Excessive values can lead to overfitting or unnecessary computations.
```

File: content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+- `random_state`: Specifies the random seed for reproducibility. The default value is `None`.
```
Comment (PragatiVerma18) [inline content/sklearn/concepts/stochastic-gradient-descent/stochastic-gradient-descent.md:61]: While `random_state` is mentioned, its importance in ensuring reproducibility is not explicitly highlighted. Include it as a note or as follows -

```suggestion
- `random_state`: Specifies the random seed for reproducibility. The default value is `None`. Setting `random_state` ensures consistent results across runs by fixing the randomness of data splitting or model initialization.

```

**Review by PragatiVerma18 — CHANGES_REQUESTED** [review]
Hey @Sriparno08, thanks for contributing this entry. I just have a few suggestions, please go through the comments and do the required changes, and then we can merge this! 

Let me know if you have any questions. Happy contributing! 🚀

**Comment by Sriparno08** [comment]
Made the changes, @PragatiVerma18.

**Review by PragatiVerma18 — APPROVED** [review]
Hey @Sriparno08, thank you for quickly making the changes. This looks good to be merged! 🚀
